crates/bevy_time/src/commands.rs

```diff
@@ -0,0 +1,186 @@
+use std::{
+    ops::{Deref, DerefMut},
+    time::Duration,
+};
+
+use bevy_ecs::{
+    system::{Commands, Deferred, Res, ResMut, Resource, SystemBuffer, SystemMeta, SystemParam},
+    world::{CommandQueue, World},
+};
+
+use crate::{Domain, Time, Timer, TimerMode};
+
+/// A [`SystemParam`] that allows you to queue commands to be
+/// executed after a certain duration has elapsed,
+/// rather than immediately at the next sync point.
+///
+/// This type also derefs to [`Commands`], so you can still use it
+/// to queue commands to run at the next sync point.
+///
+/// **Note**: By default, commands that have elapsed will be queued
+/// in the [`First`] and [`FixedFirst`] schedules for time domains
+/// `()` and [`Fixed`], respectively.
+///
+/// # Usage
+///
+/// Add `mut commands: TimedCommands` as a function argument to your system,
+/// and call [`TimedCommands::after`] to get a [`Commands`] that will queue its commands
+/// to run after the specified duration has elapsed.
+///
+/// ```
+/// # use bevy_time::prelude::*;
+/// # use std::time::Duration;
+/// #
+/// fn my_system(mut commands: TimedCommands) {
+///     commands.after(Duration::from_secs(5))
+///         .spawn_empty();
+///
+///     commands.spawn_empty();
+/// }
+/// # bevy_ecs::system::assert_is_system(my_system);
+/// ```
+///
+/// [`First`]: bevy_app::First
+/// [`FixedFirst`]: bevy_app::FixedFirst
+/// [`Fixed`]: crate::Fixed
+#[derive(SystemParam)]
+pub struct TimedCommands<'w, 's, T: Domain = ()> {
+    commands: Commands<'w, 's>,
+    queues: Deferred<'s, TimedCommandQueues<T>>,
+}
+
+impl<'w, 's, T: Domain> TimedCommands<'w, 's, T> {
+    /// Creates a new [`Commands`] instance that will have its commands queued
+    /// after the specified duration has elapsed.
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// # use bevy_time::prelude::*;
+    /// # use std::time::Duration;
+    /// #
+    /// fn my_system(mut commands: TimedCommands) {
+    ///     commands.after(Duration::from_secs(5))
+    ///         .spawn_empty();
+    /// }
+    /// # bevy_ecs::system::assert_is_system(my_system);
+    /// ```
+    #[must_use = "no commands are queued by this method itself, only by the returned Commands"]
+    pub fn after(&mut self, duration: Duration) -> Commands<'w, '_> {
+        let timer = Timer::new(duration, TimerMode::Once);
+        let queue = CommandQueue::default();
+        self.queues.inner.push((timer, queue));
+        let (_, queue) = self
+            .queues
+            .inner
+            .last_mut()
+            .unwrap_or_else(|| unreachable!("we just pushed a queue"));
+        self.commands.with_queue(queue)
+    }
+}
+
+impl<'w, 's, T: Domain> Deref for TimedCommands<'w, 's, T> {
+    type Target = Commands<'w, 's>;
+
+    fn deref(&self) -> &Self::Target {
+        &self.commands
+    }
+}
+
+impl<'w, 's, T: Domain> DerefMut for TimedCommands<'w, 's, T> {
```

Comment:
At this point, I feel like we've effectively replaced `Commands` entirely. There's no real reason to use `Commands` as a system parameter, when `TimedCommands` is just a `Commands` with extra features.